packages/icon-button/src/IconButton.tsx

```diff
@@ -57,6 +57,12 @@ interface ButtonIconButtonProps
 
 type IconButtonProps = LinkIconButtonProps | ButtonIconButtonProps;
 
+function usesLinkElement(
```

Comment:
What's with these changes?

Reply:
Typescript started to complain about onClick handling. Have encountered this before with a component dynamically switching between 'a' and 'button' tags so just followed pattern that we've implemented in other places.

Reply:
Should run a changeset for this component, great catch.